### PR TITLE
changes to get the sample application to compile and run under windows

### PR DESCRIPTION
**Pull request by dovholuk**
Smallest delta as possible between the existing linux impl and a working windows example.

### CODE REVIEW

**Comment by ssrlive** [comment]
Well. I will delete my PR. byebye.

File: CMakeLists.txt [hunk 1]
```diff
@@ -19,15 +22,23 @@ add_subdirectory(libuv)
 set_target_properties(uv_a PROPERTIES
         INTERFACE_INCLUDE_DIRECTORIES ${CMAKE_CURRENT_SOURCE_DIR}/libuv/include)
 
-add_library(uv_mbed STATIC
-        src/uv_mbed.c src/uv_mbed.c src/bio.c src/bio.h)
+if (WIN32)
```
Comment (ekoby) [inline CMakeLists.txt]: probably better to create a list of source files and use it in `add_library()` directive, conditionally append to the list.

See libuv/CMakeLists.txt for example

File: CMakeLists.txt [hunk 2]
```diff
@@ -5,7 +5,10 @@ PROJECT("uv-mbed"
         VERSION 0.5.0
         )
 
-option(ENABLE_SAMPLES "Build samples." OFF)
+#sample is on by default
```
Comment (ekoby) [inline CMakeLists.txt]: rather have it OFF by default, then consuming projects don't have to disable it

File: sample/CMakeLists.txt [hunk 3]
```diff
@@ -1,6 +1,5 @@
-
-
 add_executable(sample sample.c)
 
-target_link_libraries(sample
-        PUBLIC uv_mbed)
\ No newline at end of file
+target_link_libraries(sample PUBLIC uv_mbed)
+message("${CMAKE_SOURCE_DIR}/include")
```
Comment (ekoby) [inline sample/CMakeLists.txt]: is this a debug leftover?

File: src/uv_mbed.c [hunk 4]
```diff
@@ -346,19 +366,25 @@ static void mbed_ssl_process_in(uv_mbed_t *mbed) {
...
+        mbed->connReq = NULL;
     }
     else if (rc == MBEDTLS_ERR_SSL_WANT_WRITE || rc == MBEDTLS_ERR_SSL_WANT_READ) {
         uv_write_t *hsw = calloc(1, sizeof(uv_write_t));
         hsw->cb = mbed_hs_write_cb;
         hsw->handle = (uv_stream_t *) mbed;
         mbed_ssl_process_out(mbed, hsw);
     }
+    else {
+        printf("WARNING: unhandled response code %d", rc);
```
Comment (ekoby) [inline src/uv_mbed.c]: use stderr for output

File: sample/sample.c [hunk 5]
```diff
@@ -66,17 +71,42 @@ void on_connect(uv_connect_t *cr, int status) {
...
+    //sets debug output if needed
+    //uv_mbed_mbedtls_debug_set_threshold(1);
+
+    mbedtls_x509_crt* ca_chain = calloc(1, sizeof(mbedtls_x509_crt));
+    uv_mbed_sample_prep_ca(ca_chain);
     uv_mbed_set_ca(&mbed, ca_chain);
 
-    uv_connect_t cr;
-    uv_mbed_connect(&cr, &mbed, "google.com", 443, on_connect);
+    mbed.connReq = calloc(1, sizeof(uv_connect_t));
```
Comment (ekoby) [inline sample/sample.c]: should not assign internal connReq here, this should be done internally

File: src/uv_mbed.c [hunk 6]
```diff
@@ -56,13 +79,12 @@ int uv_mbed_set_cert(uv_mbed_t *mbed, mbedtls_x509_crt *cert, mbedtls_pk_context
...
 int uv_mbed_connect_addr(uv_connect_t *req, uv_mbed_t* mbed, const struct addrinfo *addr, uv_connect_cb cb) {
 
-    if (mbed->_stream.connect_req != NULL && mbed->_stream.connect_req != req) {
+    if (mbed->connReq != NULL && mbed->connReq != req) {
         return UV_EALREADY;
     }
+
     req->handle = (uv_stream_t *) mbed;
     req->cb = cb;
-    mbed->_stream.connect_req = req;
```
Comment (ekoby) [inline src/uv_mbed.c]: keep assignment (to the new field)

File: src/uv_mbed.c [hunk 7]
```diff
@@ -103,9 +125,7 @@ int uv_mbed_connect(uv_connect_t *req, uv_mbed_t *mbed, const char *host, int po
     uv_getaddrinfo_t *resolve_req = malloc(sizeof(uv_getaddrinfo_t));
     req->handle = (uv_stream_t *) mbed;
     req->cb = cb;
-
-    mbed->_stream.connect_req = req;
```
Comment (ekoby) [inline src/uv_mbed.c]: need to keep assignment

File: src/uv_mbed.c [hunk 8]
```diff
@@ -10,10 +10,29 @@
 #include <assert.h>
 #include "uv_mbed/uv_mbed.h"
 #include "bio.h"
+#include "uv-common.h"
+
+#if _WIN32
```
Comment (ekoby) [inline src/uv_mbed.c]: should this be __WINDOWS__ to cover other flavors?

File: src/uv_mbed.c [hunk 9]
```diff
@@ -36,8 +55,12 @@ struct tcp_write_ctx {
 
 
 int uv_mbed_init(uv_loop_t *l, uv_mbed_t *mbed) {
-    uv__stream_init(l, (uv_stream_t *) mbed, UV_STREAM);
-
+#if _WIN32
+    uv_stream_init_dup(l, (uv_stream_t*)mbed, UV_STREAM);
```
Comment (ekoby) [inline src/uv_mbed.c]: do you still need that? don't we include internal libuv headers?

File: include/uv_mbed/uv_mbed.h [hunk 10]
```diff
@@ -8,37 +8,39 @@
...
+
     mbedtls_ssl_config ssl_config;
     mbedtls_ssl_context ssl;
 
     BIO* ssl_in;
     BIO* ssl_out;
-};
-
 
+    uv_connect_t *connReq; //a place to stash a connection request
```
Comment (ekoby) [inline include/uv_mbed/uv_mbed.h]: make this lower-snake-case for consistency

**Pull request closed without merging**